app/javascript/components/UI/rich-text/DropDownList.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
```

Comment:
Are the files in `rich-text` a replacement for `react-mentions`?

Reply:
Yes it can to replacement for react-mentions

Reply:
Using an established library is way better than writing custom solutions because it will be updated by a team of programmers without us having to do it ourselves when future changes are made to browsers and/or Javascript. Can you please break down why supporting/maintaining our own library for this feature is necessary? For example are there specific requirements that you found `react-mentions` won't be able to accommodate?

Reply:
Very good question. I can not use product which is no working how is need all and user first of all. I broke down it on the few answers although I can to write more detailed why so occured what I am not use React-mention
1. React-mention.js k during on input don't allowing made the red-violet symbol '@' as was described in my task (This punct was main why this library can not use in my task in my opinion. I try to do my work how describe in my task maximum exactly and if I can't to find special  established liibrary I will be to create my own desicion )
2. Drop-down list position have wrong no-predict position in during input
3.  during chosing names of users by using arrowUp and arrowDown  background of the select the user name was absent when have scroll
4. During input text when appearing scroll occuring overring text on the text
5. React-mention have a lot of (170) open issue did not fix still  https://github.com/signavio/react-mentions/issues and support  very slow
6. I was seek another products but didn't find anything

Reply:
@VitaliiShevchenko this is a good response, thanks. 

Some general suggestions for these kinds of situations in the future:

1. When a design requires a new component to be written because no existing component can do it, be sure to tell that to the designer what the limitations are and give them an opportunity to adjust the design requirements to make it easier to implement using an adequate existing component. If it's a design element that isn't important they will usually adjust it (at least the good designers will). 
2. After consulting the designer, if there is still a judgement call to be made between using an existing component versus creating one from scratch (for example the existing component could do the job but you still have concerns about its support level) make sure to solicit opinions of the team + managers, since it is a decision that will significantly impact maintenance long term.

Reply:
I did the opposite.
First, I hurriedly wrote a draft of the code. Further
I asked our team what they thought about it. After getting the majority's approval, I asked Mike after that. He replied that it was important for him to perform the original version.
Thanks for the advice. In the future, I will act on your advice, which will make the work easier)